nio-core/src/main/scala/zio/nio/core/package.scala

```diff
@@ -30,4 +30,11 @@ package object core {
       case _: EOFException => ZIO.none
     }
 
+  implicit final private[nio] class IOCloseableManagement[-R, +E, +A <: IOCloseable { type Env >: R }](
```

Comment:
Any particular reason to use a structured type here instead of `trait IOCloseable[Env]` ?

Reply:
When I was experimenting with blocking variants of the channel APIs I found a member type for this was less hassle, and I just copied that idea here. But for this PR, it doesn't actually matter.

Reply:
I believe using structural types uses reflection, which incurs some runtime overhead. Do you think that would matter for the usage in zio-nio?

Reply:
Reflection is definitely used to access term-level members, but I think type members are handled entirely at compile time.

Reply:
Cool, I didn't know that